registration/forms.py

```diff
@@ -40,6 +40,20 @@ class Meta:
         fields = (UsernameField(), "email")
 
 
+class RegistrationFormUsernameLowercase(RegistrationForm):
+    """
+    A subclass of :class:`RegistrationForm` which enforces unique case insensitive
+    usernames, make all usernames to lower case.
+
+    """
+    def clean_username(self):
+        username = self.cleaned_data.get('username', '').lower()
+        if User.objects.filter(username=username).exists():
```

Comment:
use `UsernameField()=username` since the username field is customizable. otherwise the code is good to go!!